src/Backend/Core/Engine/Model.php

```diff
@@ -553,12 +554,13 @@ public static function getUrlForBlock(
                     }
 
                     // loop extras
+                    /** @var ModuleExtra $extra */
                     foreach ($properties['extra_blocks'] as $extra) {
                         // direct link?
-                        if ($extra['module'] === $module && $extra['action'] === $action && $extra['action'] !== null) {
+                        if ($extra->getModule() === $module && $extra->getAction() === $action && $extra->getAction() !== null) {
                             // if there is data check if all the requested data matches the extra data
-                            if ($data !== null && isset($extra['data'])
-                                && array_intersect_assoc($data, (array) $extra['data']) !== $data
+                            if ($data !== null && $extra->unserializedData()
```

Comment:
if we unserialise and serialise it with the life cycle events we can just get data here